app/lib/frontend/email_sender.dart

```diff
@@ -24,77 +27,201 @@ void registerEmailSender(EmailSender value) =>
+    this._authClient,
+  );
 
-  /// Sends a [message] and returns when the operation completed.
-  /// Errors are only logged, they do not block the processing.
   Future<void> sendMessage(EmailMessage message) async {
-    await _update();
     final debugHeader = '(${message.subject}) '
         'from ${message.from} '
         'to ${message.recipients.join(', ')}';
-    if (_blockEmails || _server == null) {
-      _logger.info('Not sending email (SMTP not configured): '
-          '$debugHeader\n${message.bodyText}.');
-    } else {
-      _logger.info('Sending email: $debugHeader...');
-      try {
-        await retry(
-          () => send(
-            _toMessage(message),
-            _server,
-            timeout: Duration(seconds: 15),
-          ),
-          retryIf: (e) =>
-              e is TimeoutException ||
-              e is IOException ||
-              e is SmtpClientCommunicationException,
-          delayFactor: Duration(seconds: 2),
-          maxAttempts: 2,
-        );
-      } on SmtpMessageValidationException catch (e, st) {
-        _logger.info('Sending email failed: $debugHeader.', e, st);
-        throw EmailSenderException.invalid();
-      } on MailerException catch (e, st) {
-        _logger.severe('Sending email failed: $debugHeader.', e, st);
-        throw EmailSenderException.failed();
-      }
+    _logger.info('Sending email: $debugHeader...');
+    try {
+      await retry(
+        () async => send(
+          _toMessage(message),
+          await _getSmtpServer(),
+          timeout: Duration(seconds: 15),
+        ),
+        retryIf: (e) =>
+            e is TimeoutException ||
+            e is IOException ||
+            e is SmtpClientCommunicationException,
+        delayFactor: Duration(seconds: 2),
+        maxAttempts: 2,
+      );
+    } on SmtpMessageValidationException catch (e, st) {
+      _logger.info('Sending email failed: $debugHeader.', e, st);
+      throw EmailSenderException.invalid();
+    } on SmtpClientAuthenticationException catch (e, st) {
+      _logger.shout('Sending email failed due to invalid auth: $e', e, st);
+      throw EmailSenderException.failed();
+    } on MailerException catch (e, st) {
+      _logger.severe('Sending email failed: $debugHeader.', e, st);
+      throw EmailSenderException.failed();
     }
   }
 
-  Future<void> _update() async {
-    // Caching the values for 10 minutes, updating them only if the previous
-    // access happened earlier.
-    final now = DateTime.now().toUtc();
-    if (_lastUpdated != null && now.difference(_lastUpdated).inMinutes < 10) {
-      return;
+  Future<SmtpServer> _getSmtpServer() async {
+    final maxAge = DateTime.now().subtract(Duration(minutes: 15));
+    if (_accessToken == null && _accessTokenRefreshed.isBefore(maxAge)) {
+      _accessToken = _createAccessToken();
+      _accessTokenRefreshed = DateTime.now();
     }
-    final username = await secretBackend.lookup(SecretKey.smtpUsername);
```

Comment:
Please add a TODO. If these are not used anymore, we'll need to clean that up in the code and in Datastore.

Reply:
Good point!